client/app/pages/settings/OrganizationSettings.jsx

```diff
@@ -1,284 +1,105 @@
 
-import { clientConfig } from "@/services/auth";
 import recordEvent from "@/services/recordEvent";
 import OrgSettings from "@/services/organizationSettings";
-import HelpTrigger from "@/components/HelpTrigger";
 import wrapSettingsTab from "@/components/SettingsWrapper";
-import DynamicComponent from "@/components/DynamicComponent";
 
-const Option = Select.Option;
+import GeneralSettings from "./components/GeneralSettings";
+import AuthSettings from "./components/AuthSettings";
 
-class OrganizationSettings extends React.Component {
-  static propTypes = {
-    onError: PropTypes.func,
-  };
+function OrganizationSettings({ onError }) {
+  const [settings, setSettings] = useState({});
+  const [currentValues, setCurrentValues] = useState({});
+  const [isLoading, setIsLoading] = useState(true);
+  const [isSaving, setIsSaving] = useState(false);
 
-  static defaultProps = {
-    onError: () => {},
-  };
+  const onErrorRef = useRef(onError);
+  onErrorRef.current = onError;
 
-  state = {
-    settings: {},
-    formValues: {},
-    loading: true,
-    submitting: false,
-  };
-
-  componentDidMount() {
+  useEffect(() => {
     recordEvent("view", "page", "org_settings");
-    OrgSettings.get()
-      .then(response => {
-        const settings = get(response, "settings");
-        this.setState({ settings, formValues: { ...settings }, loading: false });
-      })
-      .catch(error => this.props.onError(error));
-  }
 
-  disablePasswordLoginToggle = () =>
-    !(clientConfig.googleLoginEnabled || clientConfig.ldapLoginEnabled || this.state.formValues.auth_saml_enabled);
+    let isCancelled = false;
 
-  handleSubmit = e => {
-    e.preventDefault();
-    if (!this.state.submitting) {
-      this.setState({ submitting: true });
-      OrgSettings.save(this.state.formValues)
-        .then(response => {
+    OrgSettings.get()
+      .then(response => {
+        if (!isCancelled) {
           const settings = get(response, "settings");
-          this.setState({ settings, formValues: { ...settings } });
-        })
-        .catch(error => this.props.onError(error))
-        .finally(() => this.setState({ submitting: false }));
-    }
-  };
-
-  handleChange = (name, value) => {
-    this.setState(
-      prevState => ({ formValues: Object.assign(prevState.formValues, { [name]: value }) }),
-      () => {
-        if (this.disablePasswordLoginToggle() && !this.state.formValues.auth_password_login_enabled) {
-          this.handleChange("auth_password_login_enabled", true);
+          setSettings(settings);
+          setCurrentValues({ ...settings });
+          setIsLoading(false);
+        }
+      })
+      .catch(error => {
+        if (!isCancelled) {
+          onErrorRef.current(error);
         }
+      });
+
+    return () => {
+      isCancelled = true;
+    };
+  }, []);
+
+  const handleChange = useCallback(
+    changes => {
+      setCurrentValues({ ...currentValues, ...changes });
+    },
+    [currentValues]
+  );
```

Comment:
```suggestion
  const handleChange = useCallback(changes => {
    setCurrentValues(currentValues => ({ ...currentValues, ...changes }));
  }, []);
```

Reply:
Wow, didn't know that's possible :slightly_smiling_face: Thank you!